core/contextmenu_registry.js

```diff
@@ -90,85 +170,9 @@ ContextMenuRegistry.ContextMenuOption;
 /**
  * Singleton instance of this class. All interactions with this class should be
  * done on this object.
- * @type {?ContextMenuRegistry}
- */
-ContextMenuRegistry.registry = null;
-
-/**
- * Registers a RegistryItem.
- * @param {!ContextMenuRegistry.RegistryItem} item Context menu item to
- *     register.
- * @throws {Error} if an item with the given ID already exists.
- */
-ContextMenuRegistry.prototype.register = function(item) {
-  if (this.registry_[item.id]) {
-    throw Error('Menu item with ID "' + item.id + '" is already registered.');
-  }
-  this.registry_[item.id] = item;
-};
-
-/**
- * Unregisters a RegistryItem with the given ID.
- * @param {string} id The ID of the RegistryItem to remove.
- * @throws {Error} if an item with the given ID does not exist.
- */
-ContextMenuRegistry.prototype.unregister = function(id) {
-  if (!this.registry_[id]) {
-    throw new Error('Menu item with ID "' + id + '" not found.');
-  }
-  delete this.registry_[id];
-};
-
-/**
- * @param {string} id The ID of the RegistryItem to get.
- * @return {?ContextMenuRegistry.RegistryItem} RegistryItem or null if not found
+ * @type {!ContextMenuRegistry}
  */
-ContextMenuRegistry.prototype.getItem = function(id) {
-  return this.registry_[id] || null;
-};
-
-/**
- * Gets the valid context menu options for the given scope type (e.g. block or
- * workspace) and scope. Blocks are only shown if the preconditionFn shows they
- * should not be hidden.
- * @param {!ContextMenuRegistry.ScopeType} scopeType Type of scope where menu
- *     should be shown (e.g. on a block or on a workspace)
- * @param {!ContextMenuRegistry.Scope} scope Current scope of context menu
- *     (i.e., the exact workspace or block being clicked on)
- * @return {!Array<!ContextMenuRegistry.ContextMenuOption>} the list of
- *     ContextMenuOptions
- */
-ContextMenuRegistry.prototype.getContextMenuOptions = function(
-    scopeType, scope) {
-  const menuOptions = [];
-  const registry = this.registry_;
-  Object.keys(registry).forEach(function(id) {
-    const item = registry[id];
-    if (scopeType === item.scopeType) {
-      const precondition = item.preconditionFn(scope);
-      if (precondition !== 'hidden') {
-        const displayText = typeof item.displayText === 'function' ?
-            item.displayText(scope) :
-            item.displayText;
-        /** @type {!ContextMenuRegistry.ContextMenuOption} */
-        const menuOption = {
-          text: displayText,
-          enabled: (precondition === 'enabled'),
-          callback: item.callback,
-          scope: scope,
-          weight: item.weight,
-        };
-        menuOptions.push(menuOption);
-      }
-    }
-  });
-  menuOptions.sort(function(a, b) {
-    return a.weight - b.weight;
-  });
-  return menuOptions;
-};
-
-// Creates and assigns the singleton instance.
-new ContextMenuRegistry();
+ContextMenuRegistry.registry = new ContextMenuRegistry();
 
+exports.contextMenuRegistry = ContextMenuRegistry.registry;
```

Comment:
Why are we exporting this (and for shortcut registry) now? Is it just as a shorthand for `ContextMenuRegistry.registry`? I don't like introducing API/naming changes (also needs documentation changes if we intend developers to use this) buried in what seems like a refactoring change.

I think this is a good idea, it would match the fieldRegistry and class registry patterns, but also needs a few other updates like changing the internal usages and the jsdoc for the class explaining how to use the registry.

Reply:
Good point. I've backed it out and filed https://github.com/google/blockly/issues/5950 to track and/or discuss.